src/flow/src/batching_mode/engine.rs

```diff
@@ -476,15 +476,36 @@ impl BatchingEngine {
         Ok(())
     }
 
+    /// Only flush the dirty windows of the flow task with given flow id, by running the query on it.
+    /// As flush the whole time range is usually prohibitively expensive.
     pub async fn flush_flow_inner(&self, flow_id: FlowId) -> Result<usize, Error> {
         debug!("Try flush flow {flow_id}");
+        // need to wait a bit to ensure previous mirror insert is handled
+        // this is only useful for the case when we are flushing the flow right after inserting data into it
+        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
as it turned out, it's the mirror insert(which is a async task) that happen before flush_flow in sql file, which might not complete insert into when admin flush_flow is called, this temp fix is to wait a bit so previous sql might be able to complete, proper fix need some kind of sync but it's complex